examples/platforms/nrf528xx/nrf52840/nrf52840.cmake

```diff
@@ -0,0 +1,281 @@
+#
+#  Copyright (c) 2020, The OpenThread Authors.
+#  All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are met:
+#  1. Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#  2. Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#  3. Neither the name of the copyright holder nor the
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+set(OT_PLATFORM_LIB "openthread-nrf52840" "openthread-nrf52840-transport" PARENT_SCOPE)
+
+if(NOT OT_CONFIG)
+    set(OT_CONFIG "${PROJECT_SOURCE_DIR}/examples/platforms/nrf528xx/nrf52840/openthread-core-nrf52840-config.h")
+    set(OT_CONFIG ${OT_CONFIG} PARENT_SCOPE)
+endif()
+
+set(OT_BOOTLOADER "USB" CACHE STRING "OT bootloader type")
+if(${OT_BOOTLOADER} STREQUAL "USB")
+    list(APPEND OT_PLATFORM_DEFINES "APP_USBD_NRF_DFU_TRIGGER_ENABLED=1")
+    set(LD_FILE "${PROJECT_SOURCE_DIR}/examples/platforms/nrf528xx/nrf52840/nrf52840_bootloader_usb.ld")
+elseif(${OT_BOOTLOADER} STREQUAL "UART")
+    set(LD_FILE "${PROJECT_SOURCE_DIR}/examples/platforms/nrf528xx/nrf52840/nrf52840_bootloader_uart.ld")
+elseif(${OT_BOOTLOADER} STREQUAL "BLE")
+    set(LD_FILE "${PROJECT_SOURCE_DIR}/examples/platforms/nrf528xx/nrf52840/nrf52840_bootloader_ble.ld")
+else()
+    set(LD_FILE "${PROJECT_SOURCE_DIR}/examples/platforms/nrf528xx/nrf52840/nrf52840.ld")
+endif()
+
+list(APPEND OT_PLATFORM_DEFINES
+    "OPENTHREAD_CORE_CONFIG_PLATFORM_CHECK_FILE=\"openthread-core-nrf52840-config-check.h\""
+    "OPENTHREAD_SETTINGS_RAM=0"
```

Comment:
This macro should be specified by `OT_SETTINGS_RAM`. See `etc/cmake/options.cmake`.